crates/zkvm/adapters/native/src/host.rs

```diff
@@ -0,0 +1,60 @@
+use std::{fmt, sync::Arc};
+
+use strata_zkvm::{Proof, ProofType, VerificationKey, ZkVmHost, ZkVmInputBuilder, ZkVmResult};
+
+use crate::{input::NativeMachineInputBuilder, zkvm::NativeMachine};
+
+type ProcessProofFn = dyn Fn(&NativeMachine) -> ZkVmResult<()> + Send + Sync;
+
+#[derive(Clone)]
+pub struct NativeHost {
+    pub process_proof: Arc<Box<ProcessProofFn>>,
+}
+
+impl ZkVmHost for NativeHost {
+    type Input<'a> = NativeMachineInputBuilder;
+
+    fn prove<'a>(
+        &self,
+        prover_input: <Self::Input<'a> as ZkVmInputBuilder<'a>>::Input,
+        _proof_type: ProofType,
+    ) -> ZkVmResult<(Proof, VerificationKey)> {
+        (self.process_proof)(&prover_input)?;
+        let output_ref = prover_input.output.borrow();
+
+        let output = if output_ref.is_empty() {
+            vec![]
+        } else {
+            output_ref[0].clone()
+        };
+        Ok((Proof::new(output), VerificationKey::new(vec![])))
+    }
+
+    fn get_verification_key(&self) -> VerificationKey {
+        VerificationKey::new(vec![])
+    }
```

Comment:
Since we are using empty/mock verification key it would it be simpler if we could have `VerificationKey::default()`